core/src/ast_builder/expr/function.rs

```diff
@@ -1179,4 +1198,31 @@ mod tests {
         let expected = "CAST(date AS INTEGER)";
         test_expr(actual, expected);
     }
+
+    #[test]
+    fn function_extract() {
+        let actual = col("date").extract(DateTimeField::Year);
+        let expected = "EXTRACT(YEAR FROM date)";
+        test_expr(actual, expected);
+
+        let actual = col("date").extract(DateTimeField::Month);
+        let expected = "EXTRACT(MONTH FROM date)";
+        test_expr(actual, expected);
+
+        let actual = col("date").extract(DateTimeField::Day);
+        let expected = "EXTRACT(DAY FROM date)";
+        test_expr(actual, expected);
+
+        let actual = col("date").extract(DateTimeField::Hour);
+        let expected = "EXTRACT(HOUR FROM date)";
+        test_expr(actual, expected);
+
+        let actual = col("date").extract(DateTimeField::Minute);
+        let expected = "EXTRACT(MINUTE FROM date)";
+        test_expr(actual, expected);
+
+        let actual = col("date").extract(DateTimeField::Second);
+        let expected = "EXTRACT(SECOND FROM date)";
+        test_expr(actual, expected);
```

Comment:
It looks not neccessary to have `DateTimeField` enum variant cases because this unit test is for `extract` function so it is not for testing `DateTimeField`.
It would be enough to have only a single test for this.

And then, could we have a unit test which uses `pub fn extract`?
Currently, there exists `col("date"),extract(..)` but no direct `extract` fn case.

Reply:
Oh thanks for careful review for this i changed it